### PR TITLE
chore: remove deprecated getDrawable method

### LINKED ISSUE #337: remove deprecated getDrawable method
**Describe the chore**
we use deprecated getDrawable method, we can use the resource class everywhere

**Would you like to work on the issue?**
YES

### PR DESCRIPTION
**Pull request by adityastic**
Fixes #337 

Changes: 
- Changed All deprecated functions to use resourcehelper

### CODE REVIEW

File: app/src/main/java/org/fossasia/badgemagic/adapter/SaveAdapter.kt [hunk 1]
```diff
@@ -13,9 +13,11 @@ import org.fossasia.badgemagic.R
 import org.fossasia.badgemagic.data.fragments.BadgeConfig
 import org.fossasia.badgemagic.data.fragments.ConfigInfo
 import org.fossasia.badgemagic.util.MoshiUtils
+import org.fossasia.badgemagic.util.Resource
 
 class SaveAdapter(private val context: Context?, private val list: List<ConfigInfo>, private val listener: OnSavedItemSelected) : RecyclerView.Adapter<SaveAdapter.SaveItemHolder>() {
     private var selectedPosition: Int = -1
+    private var resourceHelper = Resource()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/badgemagic/adapter/SaveAdapter.kt]: Where is this variable reassigned?

File: app/src/main/java/org/fossasia/badgemagic/ui/fragments/main_textart/MainTextArtFragment.kt [hunk 2]
```diff
@@ -69,6 +70,8 @@ class MainTextArtFragment : BaseFragment() {
     private val modeAdapter = ModeAdapter()
     private lateinit var viewModel: MainTextArtViewModel
 
+    private val resourceHelper = Resource()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/badgemagic/ui/fragments/main_textart/MainTextArtFragment.kt]: Only to be used in ViewModel. Revert to context

Comment (adityastic) [reply app/src/main/java/org/fossasia/badgemagic/ui/fragments/main_textart/MainTextArtFragment.kt]: Sure

File: app/src/main/java/org/fossasia/badgemagic/adapter/SaveAdapter.kt (hunk 1 above)
Comment (adityastic) [reply app/src/main/java/org/fossasia/badgemagic/adapter/SaveAdapter.kt]: Reasaigned? It is declared for the very first time in this class

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/badgemagic/adapter/SaveAdapter.kt]: Exactly. So why is it var?

Comment (adityastic) [reply app/src/main/java/org/fossasia/badgemagic/adapter/SaveAdapter.kt]: I can use val, I just copied the line above it and edited without noticing. You're a saviour, thanks

**Comment by adityastic** [comment]
@iamareebjamal updated

File: app/src/main/java/org/fossasia/badgemagic/ui/custom/PreviewBadge.kt [hunk 3]
```diff
@@ -32,8 +33,9 @@ private const val BUNDLE_CHECKLIST = "checkList"
 
 class PreviewBadge : View {
 
-    private var ledDisabled: Drawable
-    private var ledEnabled: Drawable
+    private val resourceHelper = Resource()
+    private var ledDisabled: Drawable = resourceHelper.getDrawable(R.drawable.ic_led) as Drawable
```
Comment (adityastic) [inline app/src/main/java/org/fossasia/badgemagic/ui/custom/PreviewBadge.kt]: forgot here too, haha stupid me

File: app/src/main/java/org/fossasia/badgemagic/adapter/SaveAdapter.kt [hunk 4]
```diff
@@ -13,9 +13,11 @@ import org.fossasia.badgemagic.R
 import org.fossasia.badgemagic.data.fragments.BadgeConfig
 import org.fossasia.badgemagic.data.fragments.ConfigInfo
 import org.fossasia.badgemagic.util.MoshiUtils
+import org.fossasia.badgemagic.util.Resource
 
 class SaveAdapter(private val context: Context?, private val list: List<ConfigInfo>, private val listener: OnSavedItemSelected) : RecyclerView.Adapter<SaveAdapter.SaveItemHolder>() {
     private var selectedPosition: Int = -1
+    private val resourceHelper = Resource()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/badgemagic/adapter/SaveAdapter.kt]: Please only use in ViewModels

Comment (adityastic) [reply app/src/main/java/org/fossasia/badgemagic/adapter/SaveAdapter.kt]: what about this -> https://github.com/fossasia/badge-magic-android/blob/development/app/src/main/java/org/fossasia/badgemagic/adapter/DrawableItemHolder.kt

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/badgemagic/adapter/SaveAdapter.kt]: Use view.context there

Comment (adityastic) [reply app/src/main/java/org/fossasia/badgemagic/adapter/SaveAdapter.kt]: Wow, that is so cool. Thanks sir

File: app/src/main/java/org/fossasia/badgemagic/ui/custom/PreviewBadge.kt [hunk 5]
```diff
@@ -32,8 +33,9 @@ private const val BUNDLE_CHECKLIST = "checkList"
 
 class PreviewBadge : View {
 
-    private var ledDisabled: Drawable
-    private var ledEnabled: Drawable
+    private val resourceHelper = Resource()
+    private val ledDisabled: Drawable = resourceHelper.getDrawable(R.drawable.ic_led) as Drawable
+    private val ledEnabled: Drawable = resourceHelper.getDrawable(R.drawable.ic_led_lit) as Drawable
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/badgemagic/ui/custom/PreviewBadge.kt]: Only use in ViewModels

Comment (adityastic) [reply app/src/main/java/org/fossasia/badgemagic/ui/custom/PreviewBadge.kt]: Updated

**Comment by iamareebjamal** [comment]
fix build

**Comment by adityastic** [comment]
@iamareebjamal fixed

**Pull request merged**